src/frontend/src/instance.rs

```diff
@@ -461,31 +461,17 @@ impl FrontendInstance for Instance {
     }
 }
 
-fn parse_stmt(sql: &str) -> Result<Statement> {
-    let mut stmt = ParserContext::create_with_dialect(sql, &GenericDialect {})
-        .context(error::ParseSqlSnafu)?;
-    // TODO(LFC): Support executing multiple SQL queries,
-    // which seems to be a major change to our whole server framework?
-    ensure!(
-        stmt.len() == 1,
-        error::InvalidSqlSnafu {
-            err_msg: "Currently executing multiple SQL queries are not supported."
-        }
-    );
-    Ok(stmt.remove(0))
+fn parse_stmt(sql: &str) -> Result<Vec<Statement>> {
+    ParserContext::create_with_dialect(sql, &GenericDialect {}).context(error::ParseSqlSnafu)
 }
 
-#[async_trait]
-impl SqlQueryHandler for Instance {
-    async fn do_query(
+impl Instance {
+    async fn query_statement(
         &self,
-        query: &str,
+        stmt: Statement,
         query_ctx: QueryContextRef,
     ) -> server_error::Result<Output> {
-        let stmt = parse_stmt(query)
-            .map_err(BoxedError::new)
-            .context(server_error::ExecuteQuerySnafu { query })?;
-
+        let query = &format!("{:?}", &stmt);
```

Comment:
I think we should avoid debug formatting the stmt at each call if possible, could we leave a todo?

sqlparser's statement implements `Display` and prints the statement as SQL, but some of our statements don't support `Display`, otherwise it'd be better to use `stmt.to_string()`.

A more efficient way is slicing the original SQL, which requires the parser telling use the indices of each SQL.

Reply:
The key issue is our statement is not `Display`, otherwise this patch could be smaller and simpler. 

Because statement is taken by its consumer, the consumers are responsible to include it in error messages. I'm going to skip this `query` string in these errors.